lib/membrane_webm/muxer.ex

```diff
@@ -0,0 +1,246 @@
+    else
+      {:ok, state}
+    end
+  end
+
+  # demand all tracks for which the last frame is not cached
+  @impl true
+  def handle_demand(:output, _size, :buffers, _context, state)
+      when state.expected_tracks == state.active_tracks do
+    demands =
+      state.tracks
+      |> Enum.filter(fn {_id, info} -> info.last_block == nil end)
+      |> Enum.map(fn {id, _info} -> {:demand, Pad.ref(:input, id)} end)
+
+    {{:ok, demands}, state}
+  end
+
+  @impl true
+  def handle_demand(:output, _size, :buffers, _context, state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_process(Pad.ref(:input, id), %Buffer{payload: data} = buffer, _context, state) do
+    timestamp = div(Buffer.get_dts_or_pts(buffer), @timestamp_scale)
+    state = put_in(state.tracks[id].last_timestamp, timestamp)
+    block = {timestamp, data, state.tracks[id].track_number, state.tracks[id].codec}
+    state = put_in(state.tracks[id].last_block, block)
+
+    if Enum.any?(state.tracks, fn {_id, info} -> info.last_block == nil end) do
+      # if there are active tracks without a cached frame then demand it
+      {{:ok, redemand: :output}, state}
+    else
+      process_next_block(state)
+    end
+  end
+
+  # the cache contains one frame for each track
+  # takes the frame that should come next and appends it to current cluster or creates a new cluster and outputs current cluster
+  defp process_next_block(state) do
+    # sort blocks according to which should be next in the cluster
+    {id, youngest_track} = hd(Enum.sort(state.tracks, &block_sorter/2))
+
+    block = youngest_track.last_block
+
+    {returned_cluster, new_cluster_acc} = step_cluster(block, state.cluster_acc)
+
+    state = put_in(state.tracks[id].last_block, nil)
+
+    if returned_cluster == nil do
+      {{:ok, redemand: :output}, %State{state | cluster_acc: new_cluster_acc}}
+    else
+      # https://www.matroska.org/technical/cues.html
+      new_cue =
+        {:CuePoint,
+         [
+           CueTime: elem(block, 0),
+           CueTrack: youngest_track.track_number,
+           CueClusterPosition: state.segment_size
+         ]}
+
+      state = update_in(state.cues, fn cues -> [new_cue | cues] end)
+      cluster_bytes = Helper.serialize({:Cluster, returned_cluster})
+      new_segment_size = state.segment_size + byte_size(cluster_bytes)
+
+      {{:ok, buffer: {:output, %Buffer{payload: cluster_bytes}}, redemand: :output},
+       %State{state | cluster_acc: new_cluster_acc, segment_size: new_segment_size}}
+    end
+  end
+
+  # Groups Blocks into Clusters.
+  #   All Block timestamps inside the Cluster are relative to that Cluster's Timestamp:
+  #   Absolute Timestamp = Block+Cluster
+  #   Relative Timestamp = Block
+  #   Raw Timestamp = (Block+Cluster)*TimestampScale
+  # Matroska RFC https://www.ietf.org/archive/id/draft-ietf-cellar-matroska-08.html#section-7-18
+  # WebM Muxer Guidelines https://www.webmproject.org/docs/container/
```

Comment:
I have also question about this because according to to Matroska documentation
```
Absolute Timestamp = Block+Cluster
Relative Timestamp = Block
Scaled Timestamp = Block+Cluster
Raw Timestamp = (Block+Cluster)*TimestampScale*TrackTimestampScale
```
But here it looks like we completely ignore `TrackTimestampScale`.

Reply:
`TrackTimestampScale` is [deprecated](https://www.ietf.org/archive/id/draft-ietf-cellar-matroska-08.html#section-8.1.4.1.17-1.20)